backend/tests/prisonerProfileService.test.ts

```diff
@@ -1113,27 +1109,23 @@ describe('prisoner profile service', () => {
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore
       curiousApi.getLearnerNeurodivergence.mockResolvedValue([neurodivergenceDataNoSupportIdentified])
-      await service.getPrisonerProfileData(context, offenderNo)
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      const neurodivergence = await curiousApi.getLearnerNeurodivergence(context, offenderNo)
-      const hasDivergenceSupport = neurodivergence.some((element) => {
-        // eslint-disable-next-line eqeqeq
-        return !(element.neurodivergenceSupport == NeurodivergenceType.NoidentifiedNeurodiversityNeed)
-      })
+      const response = await service.getPrisonerProfileData(context, offenderNo)
+      const hasIdentifiedDivergenceSupportNeed = response.neurodivergenceData[0].neurodivergenceSupport.some(
+        (element) => {
+          return element.neurodivergenceSupport === NeurodivergenceType.NoidentifiedNeurodiversityNeed
+        }
+      )
       // @ts-expect-error ts-migrate(2339) FIXME: Property 'currentUser' does not exist on type '{}'... Remove this comment to see the full error message
       expect(oauthApi.currentUser).toHaveBeenCalledWith(context)
-      expect(hasDivergenceSupport).toEqual(false)
+      expect(hasIdentifiedDivergenceSupportNeed).toEqual(false)
     })
     it('should handle the not found error', async () => {
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore
       curiousApi.getLearnerNeurodivergence.mockResolvedValue([neurodivergenceNotFound])
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      const errorResponse = await curiousApi.getLearnerNeurodivergence(context, offenderNo)
-      expect(errorResponse.neurodivergenceSupport).toEqual(undefined)
-      expect(errorResponse).toEqual([
+
+      const errorResponse = await service.getPrisonerProfileData(context, offenderNo)
+      expect(errorResponse.neurodivergenceData).toEqual([
```

Comment:
IO am a bit confused by this - you expect a 404 from the `getPrisonerProfileData` call? That doesn't sound like it has been "handled" as the test name suggests?!

Reply:
Sorted by me